utils/python/CIME/XML/env_mach_pes.py

```diff
@@ -95,6 +95,8 @@ def get_total_tasks(self, comp_classes):
             pstrid = self.get_value("PSTRID_%s"%comp)
             tt = rootpe + (ntasks - 1) * pstrid + 1
             total_tasks = max(tt, total_tasks)
+        if total_tasks < self.get_value("PES_PER_NODE"): # all PEs fit in 1 node
```

Comment:
Okay but I think that we need another way to fix this - here's why: If a case is created with a default total_tasks < PES_PER_NODE the PES_PER_NODE value is changed for the job.   If the user then changes the total_tasks for the case so that it now exceeds the cases PES_PER_NODE value they will be using less than the full capacity of the node and will be unaware that the situation exists.